drizzlepac/hapsequencer.py

```diff
@@ -1411,3 +1419,62 @@ def _verify_sci_hdrname(filename):
     if closefits:
         fhdu.close()
         del fhdu
+
+# ------------------------------------------------------------------------------
+
+
+def reset_keywords_apriori(filename):
+    """
+    Check the SCI extension(s) of the output FLT/FLC and DRZ/DRC files.  If the active
+    WCS solution is 'a priori', there should not be any valid values for NMATCHES,
+    RMS_RA/RMS_DEC, FITGEOM, and CRDER1/CRDER2 as these keywords are associated with 
+    'a posteriori' solutions.  Ensure the WCSTYPE is based upon the active WCSNAME.
+
+    Parameters
+    ----------
+    filename : str
+        Name of file to evaluate
+
+    Returns
+    -------
+    None
+    """
+    wcs_keys = {'NMATCHES': 0, 'RMS_RA': 0.0, 'RMS_DEC': 0.0, 'FITGEOM': '', 
+                'CRDER1': 0.0, 'CRDER2': 0.0}
+
+    # Determine the number of SCI extensions
+    num_sci, extname = util.count_sci_extensions(filename)
+    extname_list = [(extname, x + 1) for x in range(num_sci)]
+
+    # Open the file in anticipation of updating 'a priori' solution associated keywords
+    hdu = fits.open(filename, mode='update')
+
+    # Loop over the SCI extensions in the file
+    for ext in extname_list:
+        active_wcsname = hdu[ext].header['WCSNAME']
+
+        # An 'a priori' solution is of the form <Starting WCS>-<Astrometric Catalog>
+        # Catalogs: GSC240, HSC30, GAIAD1, GAIADR2, and GAIAeDR3
+        wcsname_list = active_wcsname.split('-')
+
+        # If the WCS solution is 'a priori' ensure any keywords in wcs_keys are set to default values
+        is_apriori = len(wcsname_list) == 2 and wcsname_list[1].upper().startswith(('GSC', 'HSC', 'GAIA'))
+        if is_apriori:
+            try:
+                # Make sure the WCSTYPE is consistent with the active 'a priori' WCSNAME
+                if hdu[ext].header['WCSTYPE']:
+                    hdu[ext].header['WCSTYPE'] = updatehdr.interpret_wcsname_type(active_wcsname)
+            except KeyError:
+                pass
```

Comment:
`WCSTYPE` is optional keyword not part of FITS WCS standard. `try-except` is used to catch the case when `WCSTYPE` does not exist. Don't you want to update/set `WCSTYPE` anyway if you know what it is? So, if `try-except` deals with missing `WCSTYPE`, what is the purpose of `if hdu[ext].header['WCSTYPE']`? What happens when `WCSTYPE` is present in the header but it is an empty string `''`?
```suggestion
        if is_apriori:
            # Make sure the WCSTYPE is consistent with the active 'a priori' WCSNAME
            if 'WCSTYPE' in hdu[ext].header:
                hdu[ext].header['WCSTYPE'] = updatehdr.interpret_wcsname_type(active_wcsname)
```